### PR TITLE
Enabling Whitebox and Terrain python tests in Linux

### LINKED ISSUE #7828: WhiteBox: test_WhiteBox_SetDefaultShape fails on Linux
**Describe the bug**
test_WhiteBox_SetDefaultShape fails when run on Linux with the following errors:

```
[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-383)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-384)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-385)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-386)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-387)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-388)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-389)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-390)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-391)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-392)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-393)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-394)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-395)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-396)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-397)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-398)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-399)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-400)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-401)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-402)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-403)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-404)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-405)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-406)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-407)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-408)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-409)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-410)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-411)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-412)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-413)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-414)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-415)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-416)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-417)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-418)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-419)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-420)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-421)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-422)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-423)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-424)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-425)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-426)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-427)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-428)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-429)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-430)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-431)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-432)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-433)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-434)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-435)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-436)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-437)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-438)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-439)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-440)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-441)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-442)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-443)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-444)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-445)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-446)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-447)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-448)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-449)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-450)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-451)[](https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline#step-714-log-452)[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (python) - Running automated test: /data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py (testcase )

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (python_test) - Info: Open level /Base

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (python_test) - 

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  Opening level levels/base/base.spawnable

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] XML reader: File is empty (@user@/Editor/EditorEventLog.xml)

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] XML reader: junk after document element at line 8 (@user@/Editor/EditorEventLog.xml)

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  Loading from /data/workspace/o3de/AutomatedTesting/Levels/Base/Base.prefab...

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  ---- Start: Game Engine level load

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  Loading map '/data/workspace/o3de/AutomatedTesting/Levels/Base' into engine...

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  ---- End: Game Engine level load (0 seconds)

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  ---- Start: Post Load

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  ---- End: Post Load (0 seconds)

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [LevelLoadTime] Level /data/workspace/o3de/AutomatedTesting/Levels/Base loaded in 0 seconds

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  -----------------------------------------------------------

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  Successfully opened document /data/workspace/o3de/AutomatedTesting/Levels/Base/Base.prefab

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  Level loading time: 0.09 seconds

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  -----------------------------------------------------------

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] (AssetDatabase) - Load time threshold exceeded: LoadAssetData call for translationassets/types/ondemandreflectedtypes.names took 218 ms

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (RHI) - ****************************************************************

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (RHI) -                     Registering null RHI                          

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  (RHI) - ****************************************************************

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] (AuxGeom) - Failed to allocate dynamic buffer of size 24.

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] (Prefab Serialization) - The target path for "remove" operation has an invalid index.

[2022-02-22T20:57:51.582Z] E   [editor_test.log]      Reason: The operation has halted during a merge operation for '/1' because a field or element has an invalid value.

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Error] (Prefab) - Some of the patches were not successfully applied.

[2022-02-22T20:57:51.582Z] E   [editor_test.log]  [Warning] (python) - Serialize generic class info name:unsigned long version:0 isContainer:false)

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python) - A behavior class is missing for {5EC2D6F7-6859-400F-9215-C106F5B10E53}!

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Warning] (python) - Failed to allocate return value for method WhiteBoxMeshHandle::MeshFaceCount

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (python_test) - Failure: Default shape is not set to cube

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (python_test) - 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Warning] (python) - Serialize generic class info name:unsigned long version:0 isContainer:false)

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python) - A behavior class is missing for {5EC2D6F7-6859-400F-9215-C106F5B10E53}!

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Warning] (python) - Failed to allocate return value for method WhiteBoxMeshHandle::MeshFaceCount

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (python_test) - Test C29279329_WhiteBox_SetDefaultShape finished.

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  Report:

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [FAILED ] Failure: Default shape is not set to cube

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  EXCEPTION raised:

[2022-02-22T20:57:51.583Z] E   [editor_test.log]    Traceback (most recent call last):

[2022-02-22T20:57:51.583Z] E   [editor_test.log]      File "/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py", line 308, in start_test

[2022-02-22T20:57:51.583Z] E   [editor_test.log]        test_function()

[2022-02-22T20:57:51.583Z] E   [editor_test.log]      File "/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py", line 63, in C29279329_WhiteBox_SetDefaultShape

[2022-02-22T20:57:51.583Z] E   [editor_test.log]        white_box_handle.MeshFaceCount() == expected_results.get(shape_types.CUBE))

[2022-02-22T20:57:51.583Z] E   [editor_test.log]      File "/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py", line 37, in check_shape_result

[2022-02-22T20:57:51.583Z] E   [editor_test.log]        Report.critical_result(critical_shape_check,  face_count > 0)

[2022-02-22T20:57:51.583Z] E   [editor_test.log]    TypeError: '>' not supported between instances of 'NoneType' and 'int'

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  Test result:  FAILURE

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  JSON_START({"name": "WhiteBox_SetDefaultShape", "success": false, "exception": "Traceback (most recent call last):\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py\", line 308, in start_test\n      test_function()\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py\", line 63, in C29279329_WhiteBox_SetDefaultShape\n      white_box_handle.MeshFaceCount() == expected_results.get(shape_types.CUBE))\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py\", line 37, in check_shape_result\n      Report.critical_result(critical_shape_check,  face_count > 0)\n  TypeError: '>' not supported between instances of 'NoneType' and 'int'", "output": "Test C29279329_WhiteBox_SetDefaultShape finished.\nReport:\n[FAILED ] Failure: Default shape is not set to cube\nEXCEPTION raised:\n  Traceback (most recent call last):\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py\", line 308, in start_test\n      test_function()\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py\", line 63, in C29279329_WhiteBox_SetDefaultShape\n      white_box_handle.MeshFaceCount() == expected_results.get(shape_types.CUBE))\n    File \"/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py\", line 37, in check_shape_result\n      Report.critical_result(critical_shape_check,  face_count > 0)\n  TypeError: '>' not supported between instances of 'NoneType' and 'int'\nTest result:  FAILURE"})JSON_END

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (python_test) - 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - Traceback (most recent call last):

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) -   File "/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py", line 95, in <module>

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) -     

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - Report.start_test(C29279329_WhiteBox_SetDefaultShape)

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) -   File "/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/utils.py", line 317, in start_test

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) -     

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - assert success, f"Test {test_function.__name__} failed"

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - AssertionError

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - : 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - Test C29279329_WhiteBox_SetDefaultShape failed

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - 

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Warning] (python) - Detected script failure in Python script(/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py); return code -1!

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  [Error] (python_test) - EXCEPTION: Detected script failure in Python script(/data/workspace/o3de/AutomatedTesting/Gem/PythonTests/WhiteBox/tests/WhiteBox_SetDefaultShape.py); return code -1!

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - Called Terminate() with exit code: 0xf

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - AZ::Debug::Trace::Terminate(int) (+0x47) [0x7f08a1ef01d7]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - CCryEditApp::RunInitPythonScript(CEditCommandLineInfo&) (+0x1021) [0x7f08a176ce71]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - CCryEditApp::InitInstance() (+0x12ce) [0x7f08a176e70e]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - CryEditMain (+0x631) [0x7f08a1775341]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - main (+0xda) [0x55cbd5051fea]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - __libc_start_main (+0xf3) [0x7f08a5b680b3]

[2022-02-22T20:57:51.583Z] E   [editor_test.log]  (Exit) - _start (+0x2e) [0x55cbd5051e4e]
```

https://jenkins.build.o3de.org/blue/organizations/jenkins/O3DE/detail/PR-7824/1/pipeline

**Steps to reproduce (Jenkins)**
1. Update ```set(PAL_TRAIT_WHITEBOX_TESTS_SUPPORTED FALSE)``` to ```set(PAL_TRAIT_WHITEBOX_TESTS_SUPPORTED TRUE)``` in AutomatedTesting/Gem/PythonTests/WhiteBox/Platform/Linux/PAL_linux.cmake
2. Push the changes to a remote branch, and run Linux tests on the remote branch

**Expected behavior**
Tests succeed.

**Actual behavior**
Test fails with the main cause appearing to be the following test failure:

```
[FAILED ] Failure: Default shape is not set to cube
```

**Found in Branch**
development@d16a742b8ade1b1e0d20aefd877cf106150fc2a7

### PR DESCRIPTION
**Pull request by moraaar**
## What does this PR do?

Whitebox and Terrain failed in Linux for the same reason, a reflected function returned a `size_t`. It's recommend avoiding reflecting or serializing types like `size_t` or `long` because they are different per platform, leading to unexpected errors and getting in the way to do cross-platform asset processing eventually (for example processing windows assets using a linux machine), which can be something interesting in the future.

Fixes #7828

## How was this PR tested?

Run tests on windows and linux:

````
        Start 185: AutomatedTesting::WhiteBoxTests.main::TEST_RUN
27/131  Test #185: AutomatedTesting::WhiteBoxTests.main::TEST_RUN ..........................   Passed   14.49 sec
        Start 195: AutomatedTesting::TerrainTests_Main.main::TEST_RUN
129/131 Test #195: AutomatedTesting::TerrainTests_Main.main::TEST_RUN ......................   Passed   19.41 sec
````

### CODE REVIEW

**Review by hultonha — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h [hunk 1]
```diff
@@ -71,11 +71,11 @@ namespace Physics
 
         //! Returns the height field gridsize columns.
         //! @return the size of the grid in the x direction.
-        virtual size_t GetHeightfieldGridColumns() const = 0;
```
Comment (greerdv) [inline Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h:74]: Probably too late to change this but u64 seems a bit excessive here - that would be enough to do a terrain the size of the solar system at nanometre resolution.

**Review by greerdv — APPROVED** [review]

File: Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h (hunk 1 above)
Comment (moraaar) [reply Code/Framework/AzFramework/AzFramework/Physics/HeightfieldProviderBus.h:74]: That might be the case, I was being conservative here, didn't want to make any assumptions. Although in C++ it's not specified the maximum size of a `size_t`, in general it'll be 64bits on 64-bits systems. I went for the safest approach and use the same size with `AZ::u64`.

**Review by SergeyAMZN — APPROVED** [review]

**Review by mbalfour-amzn — APPROVED** [review]
Nit: In general, I think these days we prefer "uint64_t" to "AZ::u64", but I'm ok with it either way. Thanks for finding and fixing this!

**Comment by moraaar** [comment]
> Nit: In general, I think these days we prefer "uint64_t" to "AZ::u64", but I'm ok with it either way. Thanks for finding and fixing this!

Interesting, I thought that if there is a AZ version of the type that was preferred because it can be assumed it'll be handled correctly for each platform.

**Comment by lumberyard-employee-dm** [comment]
> > Nit: In general, I think these days we prefer "uint64_t" to "AZ::u64", but I'm ok with it either way. Thanks for finding and fixing this!
> 
> Interesting, I thought that if there is a AZ version of the type that was preferred because it can be assumed it'll be handled correctly for each platform.

My non-opinion on this is that AZ::s64 and AZ::u64 versions are better to use, because they are consistent across all of O3DE platforms.

They are always a `long long` or `unsigned long long`.
So they sidestep any issues with 64-bit type differences such as `long` being a different type than `long long` on Linux despite both being 64 bit.

Several issues come up such as platforms issues with using `strtoll` vs `strtol`, using printf format specifiers of `%lu` vs `%llu`.

Using AZ::u64, means that `strtoll` is the only function that needs to be used and `%llu` is the only format specifier that needs to be used.
Though we do have support for [std::from_chars and std::to_chars](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/AZStd/Charconv.cpp) which is the preferred way to convert integers <-> string.

**Pull request merged**